k9mail/src/main/java/security/planck/ui/passphrase/PassphraseManagementDialog.kt

```diff
@@ -0,0 +1,118 @@
+package security.planck.ui.passphrase
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.appcompat.app.AppCompatActivity
+import androidx.compose.material.MaterialTheme
+import androidx.compose.ui.platform.ComposeView
+import androidx.fragment.app.DialogFragment
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.FragmentManager
+import androidx.fragment.app.viewModels
+import com.fsck.k9.ui.getEnum
+import com.fsck.k9.ui.putEnum
+import dagger.hilt.android.AndroidEntryPoint
+import security.planck.ui.passphrase.unlock.compose.PassphraseUnlockDialogContent
+import security.planck.ui.passphrase.manage.PassphraseManagementViewModel
+import security.planck.ui.passphrase.manage.compose.PassphraseManagementDialogContent
+import security.planck.ui.passphrase.unlock.PassphraseUnlockViewModel
+import kotlin.system.exitProcess
+
+private const val TAG = "security.planck.passphrase.PassphraseManagementDialog"
+private const val ARG_MODE = "security.planck.passphrase.PassphraseManagementDialog.Mode"
+
+@AndroidEntryPoint
+class PassphraseManagementDialog : DialogFragment() {
+    private val managementViewModel: PassphraseManagementViewModel by viewModels()
+    private val unlockViewModel: PassphraseUnlockViewModel by viewModels()
+    private val mode: PassphraseDialogMode
+        get() = requireArguments().getEnum<PassphraseDialogMode>(
+            ARG_MODE
+        )
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        if (savedInstanceState == null) {
+            when (mode) {
+                PassphraseDialogMode.MANAGE -> managementViewModel.start()
+                PassphraseDialogMode.UNLOCK -> unlockViewModel.start()
+            }
+        }
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        val composeView = ComposeView(requireContext()).apply {
+            setContent {
+                MaterialTheme {
+                    when (mode) {
```

Comment:
ok so we have one dialog to handle both unlock and management operations, but from the same view. I thought we would have 2 distinct views, but I think it's also fine like this, if there is enough to put in common

Reply:
I mean, many things are common yes. Just it is too much code  creating a new fragment for two cases that are very similar.
Yes, we choose the view content depending on the mode we are in. All the state is held in the ViewModel anyway.
Btw this was a first Compose screen in the app and I expect it to be improved eventually.  Looks are good but it could be a bit more standarized toward Compose recommended behaviors etc and improved.

Reply:
sounds reasonable